lib/src/exceptions.dart

```diff
@@ -191,13 +191,15 @@ ServerException buildExceptionFromErrorFields(List<ErrorField> errorFields) {
 }
 
 PgException transformServerException(ServerException ex) {
-  if (ex.code == '57014') {
-    return _PgQueryCancelledException(
-      ['${ex.code}:', ex.message, ex.trace].whereType<String>().join(' '),
-      severity: ex.severity,
-    );
-  }
-  return ex;
+  return switch (ex.code) {
+    '23505' => DuplicateKeyException(ex.message, severity: ex.severity),
+    '23503' => ForeignKeyViolationException(ex.message, severity: ex.severity),
+    '57014' => _PgQueryCancelledException(
+        ['${ex.code}:', ex.message, ex.trace].whereType<String>().join(' '),
```

Comment:
wasn't sure if you want the same error message structure for all of these? i.e.:
`['${ex.code}:', ex.message, ex.trace].whereType<String>().join(' ')`

Reply:
The current version is great. Cancelled exception is different as hopefully the stacktrace provides information on where the cancellation was initiated, while the others don't need dual stack information.